### ISSUE TITLE
Create hidden "garbage bin" category, move threads to it before deleting them

### ISSUE THREAD

**Issue by rafalp** [rafalp/Misago#976]
Bulk thread deletion may be costful and slow if moderator deletes thread with plenty of posts, but this can be improved by changing thread deletion action to actually move the thread to hidden category invisible to all users instead of deleting it on HTTP thread.

This category could then be pruned automatically by either cron job or celery task.

**Comment by AetherUnbound** [comment]
In order to accomplish this, would it be [this function](https://github.com/rafalp/Misago/blob/503a11baced6a656ed152eb8e9ed5890d0040705/misago/threads/moderation/threads.py#L187-L194) that would need to be modified to move the thread to that garbage category first instead of deleting?

And would something like [this](https://github.com/rafalp/Misago/blob/fafa7d87502699f3e8d384bde644a210a6d938b5/misago/users/admin/tasks.py#L9) be an example of a celery task that this might be based on?

**Comment by rafalp** [comment]
Yup, `delete_thread` should use the `thread.move()` instead of `thread.delete()`. As for creating celery task, you can take inspiration from that one, yes. Even simpler would be to abuse [`prunethreads`](https://github.com/rafalp/Misago/blob/master/misago/categories/management/commands/prunecategories.py) management command by setting `prune_replied_after` for garbage bin category. ;)

**Comment by rafalp** [comment]
I'll do this for 0.40 because its better to delete attachments fastly, but moving threads to a garbage can category first and deleting them with a Celery beat task will be faster than waiting for the mod action to do its thing.

**Comment by rafalp** [comment]
This optimization is not as useful as I’ve thought, as the main pain point in deleting content is deleting posts that users have made in other users threads, and here there is no way around it but iterate thread by thread. Same with polls, likes and edits history.

**Comment by rafalp** [comment]
I've did `delete_post` and `delete_thread` utils for moderation actions and found that thread and post deletion is quite fast. This optimization is no longer required.
